feat(docs): menu component theming docs

hwarnke21 opened the pull request:
<!---
Thanks for creating a Pull Request 💖!

Please read the following before submitting:
- PRs that adds new external dependencies might take a while to review.
- Keep your PR as small as possible.
- Limit your PR to one type (docs, feature, refactoring, ci, or bugfix)
-->

Closes #1052 

## 📝 Description

This PR adds new documentation for theming the `Menu` component.

## ⛳️ Current behavior (updates)

Currently, there isn't any documentation for theming the `Menu` component.

## 🚀 New behavior

Adds documentation and Sandpack code example for theming the `Menu` component.

## 💣 Is this a breaking change (Yes/No):

No. 

## 📝 Additional Information
I modeled the documentation on the recent changes made to the [`Input` component theming docs](https://chakra-ui.com/docs/components/input). For the component theme code example, I followed the conventions recommended by [Component Theming: Styling multipart components](https://chakra-ui.com/docs/styled-system/component-style#styling-multipart-components).

nikolovlazar reviewed (changes requested):
@hwarnke21 we've updated the multipart component theming doc template to provide more value to the users by including more use cases and guides. Please check out the new structure and optimize your PR according to it.

hwarnke21 commented:
@nikolovlazar Thank you! Where can I find the multipart component theming doc template? I tried searching through the repo and doc site, but I didn't see anything.

nikolovlazar commented:
@hwarnke21 it's the Input one, that's included in #1052. Here's the link again: https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/input/theming.mdx

hwarnke21 commented:
@nikolovlazar Oooh, got it! This is really nice. I'll get mine updated to match.

hwarnke21 commented:
@nikolovlazar I ran into some issues attempting to recreate what you did with the input theming that might require some updates to the documentation/code examples. I'll try to explain the best that I can: 

When I originally started the theming docs, I created a fresh React app to create and test my code examples. I followed the directions in [Create React App](https://chakra-ui.com/getting-started/cra-guide) and used `npx create-react-app my-app --template @chakra-ui`. I also installed `@chakra-ui/theme-tools`. For reference, I added a public repo of that project [here](https://github.com/hwarnke21/explore-react-chakra-component-theming). 

I added your input component theme code (`theme/components/Input.ts`) from the Sandpack example to use as a reference. It showed 2 issues in my IDE: 
- `theme` is assigned a value but never used 
- `getColor` has been deprecated in the latest version of `@chakra-ui/theme-tools`

I also tried to use the command that you suggested in the Input theming docs to get autocomplete in my IDE: `npx chakra-cli tokens path/to/theme.ts` (in my case, the path is `src/theme/index.js`). It prompted me to install `chakra-cli` so I did, and then tried the command again, but it did not work. It just says "Error: command tokens not found". 

So, I may have something set up incorrectly, or there might be a step missing somewhere. Everything I have done so far, I did following the existing documentation in these articles: 
- [Create React App](https://chakra-ui.com/getting-started/cra-guide)
- [Customize Theme](https://chakra-ui.com/docs/styled-system/customize-theme)
- [Component Style](https://chakra-ui.com/docs/styled-system/component-style)

nikolovlazar commented:
Good points @hwarnke21! I've cleaned up the sandbox contents and also added a [CLI page](https://chakra-ui.com/docs/styled-system/cli) that explains in more depth how to use the CLI. Thanks for pointing these out 🙌 You can merge `main` into your branch to get the latest changes.

nikolovlazar commented on content/docs/components/menu/theming.mdx:
```diff
@@ -2,3 +2,332 @@
+
+## Anatomy
+
+<ol type='A'>
+  <li>`button`</li>
+  <li>`list`</li>
+  <li>`item`</li>
+  <li>`groupTitle`</li>
+  <li>`command`</li>
+  <li>`divider`</li>
+</ol>
```
Could you turn these into bullet points? I've realized the left padding looks pretty weird when using `<ol>`, so I updated the Input theming doc to fix that.

nikolovlazar commented on content/docs/components/menu/theming.mdx:
````diff
@@ -2,3 +2,332 @@
+
+```
+
+Every time you add anything new to the theme, you need to run the cli command to
+get proper autocomplete in your IDE. Here's how you can do that:
+
+```bash
+# run it once:
+npx chakra-cli tokens path/to/theme.ts
+
+# or run it in watch mode:
+npx chakra-cli tokens path/to/theme.ts --watch
+```
````
There's a new CLI block in the Input theming doc that you can carry over.

nikolovlazar reviewed (changes requested):
Just a couple of smaller details. Everything else looks good 🙌

nikolovlazar reviewed (approved):
Looking good, @hwarnke21! Let's ship this 🙌 There are some indentation issues in the sandbox, but I'll fix them quickly. Everything else looks good. Thank you for contributing to the Menu Theming docs!
